daemons/controld/controld_join_dc.c

```diff
@@ -586,11 +591,9 @@ do_dc_join_finalize(long long action,
         return;
     }
 
-    if ((max_generation_from != NULL)
-        && !pcmk_is_set(controld_globals.fsa_input_register, R_HAVE_CIB)) {
+    if (sync_from_max_gen) {
```

Comment:
Don't we still have to check max_generation_from != NULL? If it's NULL, it'll hit the pcmk__str_eq() above, which means sync_from_max_gen remains true, but formerly we would skip NULL

Reply:
Whoops thanks, I kept thinking I was forgetting something but then I listed out the cases and was satisfied. `pcmk__str_null_matches` is what I was forgetting.

Reply:
* `max_generation_from == NULL` -> `pcmk__str_eq()` true -> set `R_HAVE_CIB` -> fail this if-condition
* `max_generation_from != NULL` and string matches -> `pcmk__str_eq()` true -> set `R_HAVE_CIB` -> fail this if-condition
* `max_generation_from != NULL` and string doesn't match -> `pcmk__str_eq()` false -> don't set `R_HAVE_CIB` -> pass this if-condition

Just need to invert everything. Default `sync_from_max_gen = false`, set to true if `(max_generation_from != NULL)` and string doesn't match, otherwise set `R_HAVE_CIB`.

Reply:
Well that just means the `R_HAVE_CIB` check was sufficient all along. We don't need the `sync_from_max_gen` variable; we can just throw out the `(max_generation_from != NULL)` part of this if-condition. If `R_HAVE_CIB` is set, then we know `max_generation_from != NULL`.